src/emulator/gdbstub/src/gdb.cpp

```diff
@@ -258,46 +293,74 @@ static std::string cmd_read_registers(HostState &state, PacketCommand &command)
     return stream.str();
 }
 
-static std::string cmd_write_registers(HostState &state, PacketCommand &command) { return ""; }
+static std::string cmd_write_registers(HostState &state, PacketCommand &command) {
+    if (state.gdb.current_thread == -1)
+        return "";
+    CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
+
+    const std::string content = content_string(command).substr(1);
+
+    for (uint32_t a = 0; a < content.size() / 8; a++) {
+        uint32_t value = parse_hex(content.substr(a * 8, 8));
+        modify_reg(cpu, a, value);
+    }
+
+    return "OK";
+}
 
 static std::string cmd_read_register(HostState &state, PacketCommand &command) {
     if (state.gdb.current_thread == -1)
         return "";
     CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
 
-    std::string content = content_string(command);
+    const std::string content = content_string(command);
     int32_t reg = parse_hex(content.substr(1, content.size() - 1));
 
     return be_hex(fetch_reg(cpu, static_cast<uint32_t>(reg)));
 }
 
 static std::string cmd_write_register(HostState &state, PacketCommand &command) {
-    return "";
+    if (state.gdb.current_thread == -1)
+        return "";
+    CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
+
+    const std::string content = content_string(command);
+    uint32_t equal_index = content.find('=');
+    int32_t reg = parse_hex(content.substr(1, equal_index - 1));
+    uint32_t value = parse_hex(content.substr(equal_index + 1));
+    modify_reg(cpu, reg, value);
+
+    return "OK";
 }
 
-static std::string cmd_read_memory(HostState &state, PacketCommand &command) {
-    std::string content = content_string(command);
-    size_t pos = content.find(',');
+static bool check_memory_region(Address address, Address length, MemState &mem) {
+    const auto page_first = static_cast<uint32_t>(address / KB(4));
```

Comment:
KB(4) can be an constexpr, but it doesn't hurt really. Optional.

Reply:
This kills the whole point of `constexpr`, but how about `MemState::page_size`? I forgot that existed.

Reply:
Oh yeah. I forgot that. go with it